src/python/pants/invalidation/cache_manager.py

```diff
@@ -183,6 +201,11 @@ def create_results_dir(self, root_dir, allow_incremental):
       # If the target is valid, both directories can be assumed to exist.
       return
 
+    # If the vt is invalid, clean. Deletes both because if the stable_dir has somehow been replaced with a real dir,
+    # the relative_symlink call below raises an uncaught exception when it attempts to unlink the real directory.
+    safe_rmtree(current_dir)
+    safe_rmtree(stable_dir)
```

Comment:
IMO, it would be better to blow up in the case of a deleted/replaced results_dir symlink to expose an issue in a task, rather than to silently try to fix it.